src/cluster.c

```diff
@@ -2736,15 +2895,17 @@ void clusterSendPing(clusterLink *link, int type) {
      * faster to propagate to go from PFAIL to FAIL state. */
     int pfail_wanted = server.cluster->stats_pfail_nodes;
 
-    /* Compute the maximum totlen to allocate our buffer. We'll fix the totlen
+    /* Compute the maximum estlen to allocate our buffer. We'll fix the estlen
      * later according to the number of gossip sections we really were able
      * to put inside the packet. */
-    totlen = sizeof(clusterMsg)-sizeof(union clusterMsgData);
-    totlen += (sizeof(clusterMsgDataGossip)*(wanted+pfail_wanted));
+    estlen = sizeof(clusterMsg) - sizeof(union clusterMsgData);
+    estlen += (sizeof(clusterMsgDataGossip)*(wanted + pfail_wanted));
+    estlen += sizeof(clusterMsgPingExt) + getHostnamePingExtSize();
```

Comment:
here seems `sizeof(clusterMsgPingExt)` is needless since `getHostnamePingExtSize` has counted `sizeof(clusterMsgPingExt)` in, but it doesn't affect the integrity as `estlen` is just a hint for allocation, larger is fine.

Reply:
```
int getHostnamePingExtSize() {
    /* If hostname is not set, we don't send this extension */
    if (!myself->hostname) return 0;

    int totlen = sizeof(clusterMsgPingExt) + EIGHT_BYTE_ALIGN(strlen(myself->hostname) + 1);
    return totlen;
}
```
So the L2903 is equivalent to `estlen += sizeof(clusterMsgPingExt) + sizeof(clusterMsgPingExt) + EIGHT_BYTE_ALIGN(strlen(myself->hostname) + 1)`, but actually there's only one `clusterMsgPingExt` header and following a `clusterMsgPingExtHostname` body

Reply:
Ah, I see, it is counted twice.